developer_manual/digging_deeper/groupware/calendar_provider.rst

```diff
@@ -0,0 +1,637 @@
+                'protected' => true,
+            ],
+            [
+                'privilege' => '{DAV:}read',
+                'principal' => $this->getOwner() . '/calendar-proxy-read',
+                'protected' => true,
+            ],
+        ];
+    }
+
+The ACL defined for this calendar are returned. For the exact definitions, see the documentation of Sabre. At the time of writing this was:
+
+=============  ===============================   =====================================================
+entry          values                            description
+=============  ===============================   =====================================================
+``principal``  uri of principal                  The role or person trying to access the calendar
+``privilege``  ``{DAV:}read``, ``{DAV:}write``   Is the role allowed to read or to write
+``protected``  ``true``, ``false``               if ``true``, this rule is not allowed to change
+=============  ===============================   =====================================================
+
+.. code-block:: php
+
+    <?php
+    
+    function setACL(array $acl) {
+        throw new \Sabre\DAV\Exception\Forbidden('Setting ACL is not supported on this node');
+    }
+
+In this example, no updates of the ACL rules are allowed. Thus, an exception is thrown if the client tries to do so.
+
+.. code-block:: php
+
+    <?php
+    
+    function getSupportedPrivilegeSet() {
+        return null;
+    }
+
+The supported privileges can be overwritten by implementing this method. When returning ``null`` the default by Sabre is used which is fine for many tasks. See also the documentation of Sabre on this.
+
+Properties of the external calendar -- IProperties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally, there are some properties of a calendar to be specified. The CalDAV interface allows here for a rather generic interface. You will have to visit the details of the CalDAV standard on what properties make sense for you.
+
+.. code-block:: php
+
+    <?php
+    
+    function getProperties($properties) {
+        // A backend should provide at least minimum properties
+        return [
+            '{DAV:}displayname' => 'Dav Example Calendar: ' . $this->calendarUri,
+            '{http://apple.com/ns/ical/}calendar-color'  => '#565656',
+            '{' . Plugin::NS_CALDAV . '}supported-calendar-component-set' => new SupportedCalendarComponentSet(['VTODO', 'VEVENT']),
+        ];
+    }
+
+Here a basic stub of calendar properties are provided. It is a basic name, a color and the setting to allow both appointments (``VEVENT``) and tasks (``VTODO``) in the calendar.
+
+.. code-block:: php
+
+    <?php
+    
+    function propPatch(PropPatch $propPatch) {
+        // We can just return here and let oc_properties handle everything
+    }
+
+This method needs implementation to satisfy PHP but can be left empty as the core handles this most probably.
```

Comment:
**question:**
I think the return type should be `false` here for Sabre to pass the `propPatch` to the next provider. Please check the code if possible. Otherwise a parent::propPatch could also work.

Reply:
I am a bit lost on this one. I tried to do some research on this but did not really find something useful that I was able to understand right now.
- One hint is in the [calendar app](https://github.com/nextcloud/server/blob/7fecdf61299e7213fd857e7d89ce9b33db05c47b/apps/dav/lib/CalDAV/Calendar.php#L268-L274). There, obviously, nothing is returned (so, it would be a `void` return type).
- Also, in the Sabre code no return tye is declared.
- In the calendar app, some hint is made that calling the parent class might be required for shared calendars.